### DISCUSSION TITLE
How to use 'update_queue'

### DISCUSSION THREAD

**Discussion by vinodg912** [python-telegram-bot/python-telegram-bot#3527] [Q&A] [answered]
I'm creating a bot with gpt-3 chat memory
But I've got some error after completing code.
`Traceback (most recent call last):
  File "/data/data/com.termux/files/home/test/bot.py", line 61, in <module>
    main()
  File "/data/data/com.termux/files/home/test/bot.py", line 49, in main
    updater = Updater("*******************", update_queue)
                                                                        ^^^^^^^^^^^^`

Anyone can tell me how to fix this error

Full code here 
` import logging
import telegram
from telegram.ext import Updater, CommandHandler, MessageHandler, filters

logging.basicConfig(format='%(asctime)s - %(name)s - %(levelname)s - %(message)s',
                    level=logging.INFO)

chat_memory = {}

def update_memory(user_input, model_response):
    chat_memory[user_input] = model_response

def retrieve_response(user_input):
    if user_input in chat_memory:
        return chat_memory[user_input]
    else:
        model_response = generate_response_with_model(user_input)
        update_memory(user_input, model_response)
        return model_response

def generate_response_with_model(user_input):
    import openai
    openai.api_key = "openai_key"

    response = openai.Completion.create(
        engine="text-davinci-002",
        prompt=f"{user_input}",
        max_tokens=1024,
        n=1,
        stop=None,
        temperature=0.5,
    )

    generated_response = response["choices"][0]["text"].strip()
    return generated_response

def start(update, context):
    context.bot.send_message(chat_id=update.effective_chat.id, text="Hi, how are you")

def chat(update, context):
    user_input = update.message.text
    response = retrieve_response(user_input)
    context.bot.send_message(chat_id=update.effective_chat.id, text="Model: " + response)

def main():
    updater = Updater("bot_key", update_queue)

    dp = updater.dispatcher

    dp.add_handler(CommandHandler("start", start))
    dp.add_handler(MessageHandler(Filters.text, chat))

    updater.start_polling()

    updater.idle()

if __name__ == '__main__':
    main() `

**Comment by Poolitzer** [comment, marked as answer]
ChatGPT is not able to create code with the latest version of our library, its database only went to something like 2018 iirc. Either you downgrade to V13 or you learn how to code by yourself with V20.

**Reply by vinodg912** [reply]
Thanks :⁠-⁠)
